javascript/javascript2/week3/function_as_a_variable/function_main.js

```diff
@@ -0,0 +1,35 @@
+//Function as a variable
+
+//Create an array with 3 items. All items should be functions. Iterate through the array and call all the functions.
+const func1 = () => {
+    console.log('I am function1');
+}
+const func2 = () => {
+    console.log('I am function2');
+}
+const func3 = () => {
+    console.log('I am function3');
+}
+const arr = [func1(), func2(), func3()];
+arr.forEach((func) => {
+    return func;
+})
+
+
+//Create a function as a const and try creating a function normally. Call both functions.
+const myFunction = () => {
+    console.log('my Function');
+}
+
+function myNormalFunction() {
+    console.log('my noral Function');
+}
+myFunction();
+myNormalFunction();
+
+
+//Create an object that has a key whose value is a function. Try calling this function.
+const object = {
+    key: myFunction()
+};
+object.key;
```

Comment:
similar to the array example, you can store the funciton as a variable in an object like
```
object = {
   key: myFunction
}
```
then call it with
```
object.key()
```

this is the difference between storing the result of the function in an object and the function itself